classes/tegra-uefi-signing.bbclass

```diff
@@ -13,20 +15,42 @@ TEGRA_UEFI_SIGNING_DEPENDS ?= "${@tegra_uefi_signing_deps(d)}"
 
 # Standard signing, input file modified with signature
 tegra_uefi_sbsign() {
-    sbsign --key "${TEGRA_UEFI_DB_KEY}" --cert "${TEGRA_UEFI_DB_CERT}" --output "$1" "$1"
+    if [ -n "${TEGRA_UEFI_DB_KEY}" -a -n "${TEGRA_UEFI_DB_CERT}" ]; then
+        sbsign --key "${TEGRA_UEFI_DB_KEY}" --cert "${TEGRA_UEFI_DB_CERT}" --output "$1" "$1"
+    fi
 }
 
 # Separate signature file, for NVIDIA's L4TLauncher 
 tegra_uefi_split_sign() {
-    openssl cms -sign -signer "${TEGRA_UEFI_DB_CERT}" -inkey "${TEGRA_UEFI_DB_KEY}" -binary -in "$1" -outform der -out "$1".sig
+    if [ -n "${TEGRA_UEFI_DB_KEY}" -a -n "${TEGRA_UEFI_DB_CERT}" ]; then
+        openssl cms -sign -signer "${TEGRA_UEFI_DB_CERT}" -inkey "${TEGRA_UEFI_DB_KEY}" -binary -in "$1" -outform der -out "$1".sig
+    fi
 }
 
 # Signature attached to end, another NVIDIA special
 # Input file remains intact; output file has ".signed" suffix
 tegra_uefi_attach_sign() {
-    openssl cms -sign -signer "${TEGRA_UEFI_DB_CERT}" -inkey "${TEGRA_UEFI_DB_KEY}" -binary -in "$1" -outform der -out "$1".sig.tmp
-    cp "$1" "$1.signed"
-    truncate --size=%2048 "$1.signed"
-    cat "$1".sig.tmp >> "$1.signed"
-    rm "$1".sig.tmp
+    if [ -n "${TEGRA_UEFI_DB_KEY}" -a -n "${TEGRA_UEFI_DB_CERT}" ]; then
+        openssl cms -sign -signer "${TEGRA_UEFI_DB_CERT}" -inkey "${TEGRA_UEFI_DB_KEY}" -binary -in "$1" -outform der -out "$1".sig.tmp
+        cp "$1" "$1.signed"
+        truncate --size=%2048 "$1.signed"
+        cat "$1".sig.tmp >> "$1.signed"
+        rm "$1".sig.tmp
+    fi
```

Comment:
Based on how the `sign_bootimg()` function is calling this (up above in `image_types_cboot.bbclass`), I think you'll need to and an `else` clause here to copy `$1` to `$1.signed` if the key/cert variables are not set.

Reply:
I unconditionally copied it since both the `if` and `else` clause would have executed it.

Reply:
Right, I see now.